recipes/eastl/all/conanfile.py

```diff
@@ -25,92 +40,107 @@ class EastlConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     @property
-    def _minimum_cpp_standard(self):
+    def _min_cppstd(self):
         return 14
 
     @property
-    def _minimum_compilers_version(self):
+    def _compilers_minimum_version(self):
         return {
             "Visual Studio": "14",
-            "gcc": "5",
-            "clang": "3.2",
-            "apple-clang": "4.3",
+            "gcc": "6",
+            "clang": "6",
+            "apple-clang": "6",
         }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("eabase/2.09.06")
+        self.requires("eabase/2.09.12", transitive_headers=True, transitive_libs=True)
```

Comment:
```suggestion
        self.requires("eabase/2.09.12")
```
Not necessary for ``package_type = "header-library"``